### ISSUE TITLE
Automatically Change Streams field of data remote reindexed to match connected index stream ID(s)

### ISSUE THREAD

**Issue by mcdowellster** [Graylog2/graylog2-server#18882]
## What?
Automatically update streams field with stream id(s) connected to the index that the data lives in. 

## Why?
Remote reindexing data from one cluster to another, say when upgrading to a much more modern graylog, results in log messages stored in OpenSearch but not queryable in Graylog. Once updating this field you can search logs in graylog. 

* Graylog Version: 5.2.4
* OpenSearch Version: 2.5
* MongoDB Version: 6.0.13
* Operating System: Ubuntu 22.04

**Comment by todvora** [comment]
Hey @mcdowellster, could you give me some examples, please?

**Comment by mcdowellster** [comment]
Certainly. In the example environment remote reindexing was attempted to move data from old cluster to the new cluster. They recreated the streams without a mongo restore so they all had new IDs. 

In order to view messages in graylo after remote reindexing I had to update the log messages to reflect the stream ID on the new cluster. 

This may become an issue for users opting to do a migration with a net new cluster. Existing clusters should have no issues.

**Comment by todvora** [comment]
I think this is way outside of the migration scope. The migration can't and shouldn't do anything like that. If you start with completely new setup and empty mongodb, then datanode migrations are not the right tool for importing any "unmanaged" data. We can't correctly configure indices, index sets, templates, mappings...
